### PR TITLE
Clarification of all options available for offline mapping

### PR DESCRIPTION
**Pull request by JaffaKetchup**
There are now 4 different ways of caching/downloading/preconfiguring offline mapping, so this PR aims to describe them all, and say which option is best for which use case.
I also fixed some capitalisation inconsistencies.

### CODE REVIEW

**Comment by kengu** [comment]
@JaffaKetchup Can you change the merge to branch issues/829_nullsafety?

**Comment by JaffaKetchup** [comment]
Hi, I've changed the merge base.

**Comment by JaffaKetchup** [comment]
Hi, any news on this? Thanks 👍 .

**Comment by JaffaKetchup** [comment]
Sorry to ask again, but is there any chance of this being merged soon? Thanks :).

**Comment by JaffaKetchup** [comment]
https://github.com/fleaflet/flutter_map/pull/894#issuecomment-852968194?

**Comment by JaffaKetchup** [comment]
I have further changes this PR to reflect the new information I have found looking through the source code.

File: README.md [hunk 1]
```diff
@@ -103,7 +103,7 @@ Widget build(BuildContext context) {
 
 To use Azure Maps you will need to [setup an account and get a subscription key](https://docs.microsoft.com/en-us/azure/azure-maps/quick-demo-map-app)
 
-### Open Street Map provider
+### Open Street Maps Provider
```
Comment (johnpryan) [inline README.md:106]: I think these titles were meant to be sentence-case. https://developers.google.com/style/headings

**Review by johnpryan — APPROVED** [review]

**Comment by JaffaKetchup** [comment]
Thanks!

**Comment by aytunch** [comment]
@johnpryan On a seperate note; has MapBox ever got back to you about using them as a maps provider? I wrote them on Twitter but they haven't responded back to me.

**Comment by johnpryan** [comment]
@aytunch The latest I've heard from them is that we can't recommend MapBox, as it violates their terms of service: https://github.com/fleaflet/flutter_map/issues/404

**Comment by ibrierley** [comment]
I tried to contact them r.e vector tile stuff as well, and no reply.

**Pull request merged**